jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/CharsetStringBuilder.java

```diff
@@ -40,13 +53,30 @@ default void append(byte[] b, int offset, int length)
             append(b[i]);
     }
 
+    default void append(CharSequence chars, int offset, int length)
+    {
+        int end = offset + length;
+        for (int i = offset; i < end; i++)
+            append(chars.charAt(i));
+    }
+
     default void append(ByteBuffer buf)
     {
         int end = buf.position() + buf.remaining();
         while (buf.position() < end)
             append(buf.get());
     }
 
+    /**
+     * @return The decoded built string.
+     * @throws RuntimeException if there is a {@link CharacterCodingException}.
```

Comment:
Whats the point of this, why not just throw `CharacterCodingException` if there is a `CharacterCodingException`? And why doesn't `takeString` do the same?

Why is this called `toCompleteString` and the one below is not called `takeCompleteString`?

Reply:
OK, I renamed `takeString()` to `build()` as this is a builder after all.